arviz/data/io_pymc3.py

```diff
@@ -370,11 +385,20 @@ def predictions_from_pymc3(
         Coordinates for the variables.  Map from coordinate names to coordinate values.
     dims: Dict[str, array-like[str]]
         Map from variable name to ordered set of coordinate names.
-
+    idata_orig: InferenceData, optional
+        If supplied, then modify this inference data in place, adding ``predictions`` and
+        (if available) ``predictions_constant_data`` groups. If this is not supplied, make a
+        fresh InferenceData
     Returns
     -------
-    InferenceData
+    InferenceData:
+        May be modified ``idata_orig``.
     """
-    return from_pymc3(
+    new_idata = PyMC3Converter(
         trace=posterior_trace, predictions=predictions, model=model, coords=coords, dims=dims
-    )
+    ).to_inference_data()
+    if idata_orig is None:
+        return new_idata
+    idata_orig.predictions = new_idata.predictions
+    idata_orig.predictions_constant_data = new_idata.predictions_constant_data
```

Comment:
I would use concat for two reasons: the first is that this would allow to use `**kwargs` and pass them to `concat` (which already has both inplace and copy arguments), the second is that this does not update the `_groups` attribute (used to iterate over groups in several cases such as saving or printing)

Reply:
OK, I will change that and push.